### PR TITLE
This allows users to use the "client_credentials" grant type.

### PR DESCRIPTION
**Pull request by thomasliasn**

### CODE REVIEW

File: src/main/java/org/zalando/stups/tokens/AccessTokenRefresher.java [hunk 1]
```diff
@@ -82,7 +82,10 @@ void start() {
         initializeFixedTokensFromEnvironment();
         LOG.info("Starting to refresh tokens regularly...");
         final ClientCredentials clientCredentials = configuration.getClientCredentialsProvider().get();
-        final UserCredentials userCredentials = configuration.getUserCredentialsProvider().get();
+        UserCredentials userCredentials = null;
+        if(configuration.getUserCredentialsProvider() != null) {
```
Comment (hjacobs) [inline src/main/java/org/zalando/stups/tokens/AccessTokenRefresher.java]: could you add a space before "("?

**Comment by bomgar** [comment]
Fixed formatting

**Comment by schlomo** [comment]
Hi,

we would really appreciate it if you could merge this PR.

Kind Regards,
Schlomo
ImmobilienScout24

**Comment by jbellmann** [comment]
Thank you for contributing.

**Pull request merged**